v4l2rtspserver-master/inc/font/BaseFont.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef BASEFONT_H
+#define BASEFONT_H
+
+typedef struct bitmapinfo {
+    uint32_t width;
+    uint32_t *pdata;
+} bitmapinfo_t;
+
+class BaseFont {
+    public:
+        bitmapinfo_t *character_map;
+
+        char start_char;
+        char end_char;
+        int height;
+
+        BaseFont(bitmapinfo_t *character_map_, char start_char_, char end_char_, int height_):
+            character_map(character_map_),
+            start_char(start_char_),
+            end_char(end_char_),
+            height(height_) {}
+
+
+    public:
+        bool isSupported(char c) const {
+            if (c == ' ') {
+                return true;
+            }
+
+            return (start_char <= c) && (c <= end_char);
+        }
+
+        int getWidth(char c) const {
+            if ((c == ' ') && (c < start_char)) {
```

Comment:
why not using isSupported here ?

Reply:
I hard-coded spaces into both of the sans-serif fonts so this isn't necessary any more.